sql/planbuilder/scope.go

```diff
@@ -41,9 +42,17 @@ type scope struct {
 	windowDefs  map[string]*sql.WindowDefinition
 	// exprs collects unique expression ids for reference
 	exprs map[string]columnId
+	proc  *procCtx
 }
 
 func (s *scope) resolveColumn(table, col string, checkParent bool) (scopeColumn, bool) {
+	// procedure params take precedence
+	if table == "" && checkParent && s.procActive() {
```

Comment:
Same basic comment as above

If the rules for resolving columns varies so much in the context of evaluating stored procedures then there should maybe be mostly divergent code paths for them